classes/oai/ojs/OAIDAO.inc.php

```diff
@@ -246,6 +196,74 @@ function setOAIData($record, $row, $isRecord = true) {
 
 		return $record;
 	}
+
+	/**
+	 * Get a OAI records record set.
+	 * @param $setIds array Objects ids that specify an OAI set,
+	 * in hierarchical order.
+	 * @param $from int/string *nix timestamp or ISO datetime string
+	 * @param $until int/string *nix timestamp or ISO datetime string
+	 * @param $set string
+	 * @param $submissionId int optional
+	 * @param $orderBy string UNFILTERED
+	 * @return ADORecordSet
+	 */
+	function _getRecordsRecordSet($setIds, $from, $until, $set, $submissionId = null, $orderBy = 'journal_id, submission_id') {
+		$journalId = array_shift($setIds);
+		$sectionId = array_shift($setIds);
+
+		$params = array();
+		if ($journalId) $params[] = (int) $journalId;
+		if ($sectionId) $params[] = (int) $sectionId;
+		if ($submissionId) $params[] = (int) $submissionId;
+		if ($journalId) $params[] = (int) $journalId;
+		if ($sectionId) $params[] = (int) $sectionId;
+		if (isset($set)) $params[] = $set;
+		if ($submissionId) $params[] = (int) $submissionId;
+		$result = $this->retrieve(
+			'SELECT	LEAST(a.last_modified, i.last_modified) AS last_modified,
+				a.submission_id AS submission_id,
+				j.journal_id AS journal_id,
+				s.section_id AS section_id,
+				i.issue_id,
+				NULL AS tombstone_id,
+				NULL AS set_spec,
+				NULL AS oai_identifier
+			FROM
+				published_submissions pa
+				JOIN submissions a ON (a.submission_id = pa.submission_id)
+				JOIN issues i ON (i.issue_id = pa.issue_id)
+				JOIN sections s ON (s.section_id = a.section_id)
+				JOIN journals j ON (j.journal_id = a.context_id)
+			WHERE	i.published = 1 AND j.enabled = 1 AND a.status <> ' . STATUS_DECLINED . '
+				' . ($journalId?' AND j.journal_id = ?':'') . '
+				' . ($sectionId?' AND s.section_id = ?':'') . '
+				' . ($from?' AND LEAST(a.last_modified, i.last_modified) >= ' . $this->datetimeToDB($from):'') . '
```

Comment:
@bozana, I think you're thinking of `MIN`/`MAX` rather than `GREATEST`/`LEAST`. The `getEarliestDatestamp` feature uses the sort order to fetch the single oldest record, using the `$orderBy` parameter to this function.